omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,236 @@
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			}
+		}
+		
+		return description;
+	}
+	
+	@PropertyGetter("location")
+	public String getLocation(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getLocation().getUuid();
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getIdentifierType(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("source")
+	public String getSource(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getSource().getUuid();
+	}
+	
+	@PropertyGetter("uuid")
+	public String getUuid(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("identifierType");
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		String locationUuid = postBody.get("location").toString();
+		String manualEntryEnabled = postBody.get("manualEntryEnabled").toString();
+		String automaticGenerationEnabled = postBody.get("automaticGenerationEnabled").toString();
+		String sourceUuid = postBody.get("source").toString();
+		String identifierUuid = postBody.get("identifierType").toString();
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		
+		if ((locationUuid != null && manualEntryEnabled != null && automaticGenerationEnabled != null && sourceUuid != null
```

Comment:
Use multiple if statements for the parameters or the ternary operator. Right now If one of the parameters is null nothing will be set.

Reply:
@adamsdenniskariuki in creating an Autogeneration option the only optional parameter is sources, all the rest are required

Reply:
ok

Reply:
@kwahalf can you prove your statement that the only optional one is sources?

Reply:
<img width="568" alt="screen shot 2017-10-09 at 11 00 31" src="https://user-images.githubusercontent.com/28532877/31329110-36ac492a-ace1-11e7-8dbb-c03aad8904c7.png">

Reply:
sorry it was the location that was optional

Reply:
Are you sure that it is only location?